### PR TITLE
Microchip: MEC172x: watchdog driver

### PR DESCRIPTION
**Pull request by jvasanth1**
Update watchdog driver to support MEC172x device

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

### CODE REVIEW

**Comment by jvasanth1** [comment]
Request to review this pull request

**Review by albertofloyd — APPROVED** [review]

**Review by jackrosenthal — APPROVED** [review]

File: drivers/watchdog/wdt_mchp_xec.c [hunk 1]
```diff
@@ -173,9 +191,15 @@ static int wdt_xec_init(const struct device *dev)
...
+	.regs = (struct wdt_regs *)(DT_INST_REG_ADDR(0)),
+	.girq = DT_INST_PROP_BY_IDX(0, girqs, 0),
+	.girq_pos = DT_INST_PROP_BY_IDX(0, girqs, 1),
+};
+
 static struct wdt_xec_data wdt_xec_dev_data;
 
 DEVICE_DT_INST_DEFINE(0, wdt_xec_init, NULL,
-		    &wdt_xec_dev_data, NULL,
+		    &wdt_xec_dev_data, &wdt_xec_config_0,
```
Comment (keith-zephyr) [inline drivers/watchdog/wdt_mchp_xec.c:203]: This driver is single instance, please add a build assert that the number of instances is only 1.

Example from busy_sim.c
```
BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(vnd_busy_sim) == 1,
	     "add exactly one vnd,busy-sim node to the devicetree");
```

Comment (jvasanth1) [reply drivers/watchdog/wdt_mchp_xec.c:206]: Added the build assert.

**Review by mbolivar-nordic — APPROVED** [review]
DT portions LGTM

**Review by keith-zephyr — APPROVED** [review]

**Pull request merged**